### ISSUE TITLE
Can't use win32 backend (and possibly linux) with `python -m` notation

### ISSUE THREAD

**Issue by pyhedgehog** [dschep/ntfy#126]
Symptom:
```
C:\>py -2 -m ntfy.cli -b win32 send "test ntfy message"
ERROR: Invalid backend win32
```
After some investigation I've found that `sys.argv` looks different before (at import time) and after call to main:
```
['-c', '-b', 'win32', 'send', 'test ntfy message']
['C:\\Python27\\lib\\site-packages\\ntfy\\cli.py', '-b', 'win32', 'send', 'test ntfy message']
```
 * There are error in `data.py` that assumes that `sys.argv[0]` refers to existing file. This renders unusable `ntfy` as a library for interactive session and one-liners.
 * `__init__.py` attempts to load all backends at import.

There are several possible solutions:
 * Fix `data.py` to catch "The system cannot find the file specified" exception or add `os.path.isfile` check before.
 * Change `data.py` to process attributes of `icon` only on first access (instead of import).
 * Change `__init__.py` to make backends loading lazy - only on first call to `notify`.

Should I create pull request now?

**Comment by dschep** [comment]
Good catch! Feel free to make a PR :smiley:
re: fixes --
* might be able to just change `sys.argv[0]` to `__file__`
* i like this idea. could turn `png` & `ico` into `@property`s of the icon class.
* not sure there's much benefit to this, given that in the lifecycle of a `ntfy` process it imports and immediately uses the backend module specified.

**Comment by dschep** [comment]
Closed by #127
